### ISSUE TITLE
Pytorch with spark dataframe input

### ISSUE THREAD

**Issue by bestfleer** [intel-analytics/BigDL#4508]
I need train pytorch model with spark dataframe input and cannot use dataloader. Some code are shown as follow:
`
spark = OrcaContext.get_spark_session()

train_data = spark.read.option("InferSchema", "true").option("header", "true").csv("./train_data.csv")
test_data = spark.read.option("InferSchema", "true").option("header", "true").csv("./test_data.csv")
model = LogisticRegressionModel(500)
optimizer = torch.optim.Adam(model.parameters(), lr=1e-2, weight_decay=1e-3)
criterion = nn.BCELoss()

est = Estimator.from_torch(model=model, optimizer=optimizer, loss=criterion)
est.fit(data=train_data, epochs=100, batch_size=64, feature_cols=train_data.columns[:-2], label_cols=[train_data.columns[-1]])
results = est.evaluate(data=test_data, batch_size=64, feature_cols=test_data.columns[:-2], label_cols=[test_data.columns[-1]])
`
However, I met following error
<img width="1380" alt="image" src="https://user-images.githubusercontent.com/39068182/165210538-9efaac68-6382-4cc9-a32e-3d6e4ea303e7.png">

Can you help me how to solve this problem? Thanks for your help

**Comment by hkvision** [comment]
@qiuxin2012 Please take a look.

@bestfleer Thanks for raising this issue, may I ask what's your dataframe schema?

**Comment by hkvision** [comment]
@qiuxin2012 Please take a look.

@bestfleer Thanks for raising this issue, may I ask what's your dataframe schema?

**Comment by bestfleer** [comment]
> @qiuxin2012 Please take a look.
> 
> @bestfleer Thanks for raising this issue, may I ask what's your dataframe schema?

All schema types are DoubleType
